src/main.go

```diff
@@ -63,6 +103,17 @@ func main() {
 		os.Exit(1)
 	}
 
+	var inCfg interface{}
+	if err := yaml.Unmarshal(dataIn, &inCfg); err != nil {
+		stderr("Failed to unmarshal input: %v", err)
+		os.Exit(1)
+	}
+
+	if hasUnrecognizedKeys(inCfg, reflect.TypeOf(cfg)) {
+		stderr("Unrecognized keys in input, aborting.")
+		os.Exit(1)
```

Comment:
I'm not sure either, but it seemed appropriate to fail since unrecognized fields aren't going to be preserved in the json output.  If they're in the ignition yaml input presumably it's desired in the output, failing seems appropriate.  Maybe add a flag to ignore warnings?